src/mem/external_master.hh

```diff
@@ -115,19 +124,36 @@ class ExternalMaster : public SimObject
     static std::map<std::string, Handler *> portHandlers;
 
   public:
+    /**
+      * @ingroup api_port
+      */
     ExternalMaster(ExternalMasterParams *params);
```

Comment:
I think constructors are part of the SimObject API... but I could be wrong. Thoughts @BobbyRBruce?

Reply:
I find this quite confusing. Even though it's a part of SimObject API, but it has its own signature, i.e. ExternalMasterParams being the parameter.

Reply:
Yeah... So, the constructors should actually never be used outside of the *Params::create() methods. In some sense, they shouldn't be "public"... In fact, now that I write this, they shouldn't be public! The *Params class should be a friend.

Let's not do make that change now... However, let's not include the constructor as part of the API.